src/NLog/Layouts/CompoundLayout.cs

```diff
@@ -0,0 +1,77 @@
+// 
+// All rights reserved.
+// 
+// Redistribution and use in source and binary forms, with or without 
+// modification, are permitted provided that the following conditions 
+// are met:
+// 
+// * Redistributions of source code must retain the above copyright notice, 
+//   this list of conditions and the following disclaimer. 
+// 
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution. 
+// 
+// * Neither the name of Jaroslaw Kowalski nor the names of its 
+//   contributors may be used to endorse or promote products derived from this
+//   software without specific prior written permission. 
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE 
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+// 
+
+
+namespace NLog.Layouts
+{
+    using Config;
+    using System.Collections.Generic;
+    using System.Text;
+
+    /// <summary>
+    /// A layout containig a nested layouts.
+    /// </summary>
+    [Layout("CompoundLayout")]
+    [ThreadAgnostic]
+    [AppDomainFixedOutput]
+    public class CompoundLayout : Layout
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CompoundLayout"/> class.
+        /// </summary>
+        public CompoundLayout()
+        {
+            Layouts = new List<Layout>();
+        }
+
+        /// <summary>
+        /// Gets the array of inner layouts.
+        /// </summary>
+        /// <docgen category='CSV Options' order='10' />
+        [ArrayParameter(typeof(Layout), "layout")]
+        public IList<Layout> Layouts { get; private set; }
+
+        /// <summary>
+        /// Formats the log event relying on inner layouts.
+        /// </summary>
+        /// <param name="logEvent">The log event to be formatted.</param>
+        /// <returns>A string representation of the log event.</returns>
+        protected override string GetFormattedMessage(LogEventInfo logEvent)
+        {
+            var sb = new StringBuilder();
+            foreach (var layout in Layouts)
+                sb.Append(layout.Render(logEvent));
+            return sb.ToString();
+        }
+    }
```

Comment:
I initially implemented InitializeLayout but later removed it:

1. CsvLayout implements it but only to set properties that ar not Lists

2. JsonLayout contains a set of attributes and does not implement it... similarly CompoundLayout contains a set of Layouts

3. When I call render each Layout is initialized

I can implement both method as a loop over Layouts that calls them for each layout